src/composable/types/twap.ts

```diff
@@ -0,0 +1,244 @@
+
+// Define the ABI tuple for the TWAPData struct
+const TWAP_DATA_ABI = [
+  'tuple(address sellToken, address buyToken, address receiver, uint256 partSellAmount, uint256 minPartLimit, uint256 t0, uint256 n, uint256 t, uint256 span, bytes32 appData)',
+]
+
+/**
+ * Parameters for a TWAP order, made a little more user-friendly for SDK users.
+ * @see {@link TWAPData} for the native struct.
+ */
+export type TWAPDataParams = Omit<TWAPData, 'partSellAmount' | 'minPartLimit'> & {
+  // total amount of sellToken to sell across the entire TWAP
+  readonly sellAmount: BigNumber
+  // minimum amount of buyToken that must be bought across the entire TWAP
+  readonly buyAmount: BigNumber
+}
+
+/**
+ * Parameters for a TWAP order, as expected by the contract's `staticInput`.
+ */
+export type TWAPData = {
+  // which token to sell
+  readonly sellToken: string
+  // which token to buy
+  readonly buyToken: string
+  // who to send the tokens to
+  readonly receiver: string
+  // amount of sellToken to sell in each part
+  readonly partSellAmount: BigNumber
+  // minimum amount of buyToken that must be bought in each part
+  readonly minPartLimit: BigNumber
+  // start time of the TWAP
+  readonly t0: BigNumber
+  // number of parts
+  readonly n: BigNumber
+  // duration of the TWAP interval
+  readonly t: BigNumber
+  // whether the TWAP is valid for the entire interval or not
+  readonly span: BigNumber
+  // appData
+  readonly appData: string
+}
+
+/**
+ * `ComposableCoW` implementation of a TWAP order.
+ * @author mfw78 <mfw78@rndlabs.xyz>
+ */
+export class TWAP extends BaseConditionalOrder<TWAPData, TWAPDataParams> {
+  /**
+   * @see {@link BaseConditionalOrder.constructor}
+   * @throws If the TWAP order is invalid.
+   * @throws If the TWAP order is not ABI-encodable.
+   * @throws If the handler is not the TWAP address.
+   */
+  constructor(
+    handler: string,
+    salt: string = utils.keccak256(ethers.utils.randomBytes(32)),
+    staticInput: TWAPDataParams,
+    hasOffChainInput = false
+  ) {
+    // First, verify that the handler is the TWAP address
+    if (handler !== TWAP_ADDRESS) throw new Error('InvalidHandler')
+
+    // Second, verify that the order params are logically valid
+    TWAP.isValid(staticInput)
+
+    // Third, construct the base class using transformed parameters
+    super(TWAP_ADDRESS, salt, staticInput, hasOffChainInput)
+
+    // Finally, verify that the transformed data is ABI-encodable
+    if (!TWAP.isValidAbi(TWAP_DATA_ABI, [this.staticInput])) throw new Error('InvalidData')
+  }
+
+  /**
+   * Create a TWAP order with sound defaults.
+   * @param staticInput The TWAP order parameters in a more user-friendly format.
+   * @returns An instance of the TWAP order.
+   */
+  static default(staticInput: TWAPDataParams): BaseConditionalOrder<TWAPData, TWAPDataParams> {
+    return new TWAP(TWAP_ADDRESS, undefined, staticInput, false)
```

Comment:
Nitpick: given that the last parameter has a default value, it could be omitted here. Unless the point is to be explicit?

```suggestion
    return new TWAP(TWAP_ADDRESS, undefined, staticInput)
```

Reply:
The 🦀 within likes explicit parameter passing 😅

If it's idiomatic with the cow-sdk to not do so, that's fine. No particular string opinion on this one.

Reply:
Not a strong opinion.
Just that, if it's optional but every invocation a parameter is passed, maybe it doesn't need to be optional?

Again, it's fine, just a nitpick

Reply:
In the spirit of lazy initialisation (and it's a handy thing in solidity), just left the default of `false` and don't explicitly declare it.